### PR TITLE
jets: non-sham dashboard

### PR DESCRIPTION
**Pull request by eamsden**
The tree-walking interpreter receives a proper jet dashboard, where `%fast` hints register cores in cold state, paths + battery axes to be jetted are written in a literal (hot state), and warm state used for matching is regenerated whenever cold state changes. 

**TODO**
- [x] sham hints are commented out. Do they stay or do they go? @philipcmonk
  * stay but disabled. We probably want a feature in the crate.
- [x] We need an interpreter context data structure. A newt, cold/warm/hot states, and (soon) persistent caches is too much to pass around as separate parameters.
- [x] Jets will need access to interpreter context to enable jetting `+mink`.
  * @ashelkovnykov I think we can roll this into your PR

### CODE REVIEW

**Comment by eamsden** [comment]
### Booting the baby pill
image

**Comment by eamsden** [comment]
@ashelkovnykov @philipcmonk this is ready for review now

**Review by ashelkovnykov — CHANGES_REQUESTED** [review]
Partial pair review w/ @eamsden

File: rust/ares/src/mem.rs [hunk 1]
```diff
@@ -468,6 +468,64 @@ impl NockStack {
         assert_no_junior_pointers!(self, *noun);
     }
 
+    pub unsafe fn struct_is_in<T>(&self, ptr: *const T, count: usize) {
```
Comment (ashelkovnykov) [inline rust/ares/src/mem.rs:471]: `assert_struct_is_in` would be a more accurate name

File: rust/ares/src/mem.rs [hunk 2]
```diff
@@ -468,6 +468,64 @@ impl NockStack {
...
+        panic!(
+            "Use after free: allocation from {:#x} to {:#x}, free space from {:#x} to {:#x}",
+            ptr as usize,
+            ptr.add(count) as usize,
+            low,
+            hi
+        );
+    }
+
+    unsafe fn noun_in(&self, noun: Noun) {
```
Comment (ashelkovnykov) [inline rust/ares/src/mem.rs:497]: TODO: Compare this to the version from the debug investigation

**Review by ashelkovnykov — CHANGES_REQUESTED** [review]

File: rust/ares/src/interpreter.rs [hunk 3]
```diff
@@ -643,10 +688,17 @@ pub fn interpret(context: &mut Context, mut subject: Noun, formula: Noun) -> Res
...
                     Todo11D::Done => {
-                        if let Some(found) =
-                            hint::match_post_nock(context, subject, dint.tag, dint.body, res)
-                        {
-                            res = found;
+                        match hint::match_post_nock(
+                            context,
+                            subject,
+                            dint.tag,
+                            Some(dint.hint),
```
Comment (ashelkovnykov) [inline rust/ares/src/interpreter.rs:695]: Need to overwrite `dint.hint` with `res` (which is actually the `clue`) in `11D::ComputeResult`

File: rust/ares/src/jets/hot.rs [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+use crate::noun::{Atom, DirectAtom, Noun, D, T};
+use ares_macros::tas;
+use either::Either::{self, Left, Right};
+use std::ptr::null_mut;
+
+const A_50: Either<u64, (u64, u64)> = Right((tas!(b"a"), 50));
+
+// This is the const state all in one spot as literals
+#[allow(clippy::complexity)]
+const HOT_STATE: &[(&[Either<u64, (u64, u64)>], u64, Jet)] = &[
```
Comment (ashelkovnykov) [inline rust/ares/src/jets/hot.rs:11]: This needs to match the sham jet dashboard at the time of merge

**Review by eamsden — COMMENTED** [review]
I approve of these changes but since I originally opened the PR I cannot officially approve them.

**Review by ashelkovnykov — APPROVED** [review]

**Pull request merged**